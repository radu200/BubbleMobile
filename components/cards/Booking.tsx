import React from "react";
import { StyleSheet, TouchableOpacity, Text, View } from "react-native";
import { formatDate, convertNumToTime } from '../../utils/time';
import Colors from "../../constants/Colors";
import StaticData from '../../data/static'
import { BookingSummary } from '../../types'


interface props {
  data: BookingSummary
}

export default function Booking({ data }: props) {

  const { hours, minutes } = convertNumToTime(1.4)

  return (
    <TouchableOpacity activeOpacity={0.9} style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.titleStatus}>
          {StaticData.booking.confirmed}
        </Text>
      </View>
      <Text style={styles.title}>
        <Text style={styles.titleLeft}>{StaticData.booking.bookingWith}: </Text> {data.otherUserFullName}
      </Text>
      <Text style={styles.title}>
        <Text style={styles.titleLeft}>{StaticData.booking.date}: </Text> {formatDate(data.scheduledStart)}
      </Text >
      <View style={styles.durationContainer}>
        <Text style={styles.titleLeft}>{StaticData.booking.duration}: </Text>
        {hours ?
          <Text style={styles.hours}>
            {hours} {StaticData.booking.hours}
          </Text> : null}
        {minutes ?
          <Text>
            {minutes} {StaticData.booking.minutes}
          </Text> : null}
      </View>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "column",
    borderRadius: 10,
    backgroundColor: Colors.light.background,
    marginBottom: 15,
    padding: 20,
    shadowColor: Colors.dark.background,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },

  header: {
    alignItems: 'center',
    marginBottom: 10,
  },

  title: {
    paddingBottom: 4,
  },

  titleStatus: {
    color: Colors.light.tint
  },

  titleLeft: {
    fontWeight: "bold"
  },

  hours: {
    marginRight: 5
  },

  durationContainer: {
    flexDirection: "row"
  }
});
