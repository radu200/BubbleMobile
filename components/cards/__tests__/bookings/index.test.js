import React from 'react';
import renderer from 'react-test-renderer';
import Booking from '../../Booking';

it('Booking component renders correctly', () => {
    const data = {
        id: "2",
        bookingStatus: true,
        scheduledStart: "2020-10-13T07:45:00.000Z",
        scheduledDuration: 8.25,
        otherUserFullName: "Janet Stevans"
    }
    const tree = renderer
        .create(<Booking data={data} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
})