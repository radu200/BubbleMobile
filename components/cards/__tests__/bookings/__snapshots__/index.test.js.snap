// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking component renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "borderRadius": 10,
      "elevation": 5,
      "flexDirection": "column",
      "marginBottom": 15,
      "opacity": 1,
      "padding": 20,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#55427A",
        }
      }
    >
      Confirmed
    </Text>
  </View>
  <Text
    style={
      Object {
        "paddingBottom": 4,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Booking With
      : 
    </Text>
     
    Janet Stevans
  </Text>
  <Text
    style={
      Object {
        "paddingBottom": 4,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Date
      : 
    </Text>
     
    October 13th 2020, 8:45 am
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Duration
      : 
    </Text>
    <Text
      style={
        Object {
          "marginRight": 5,
        }
      }
    >
      1
       
      hours
    </Text>
    <Text>
      24
       
      minutes
    </Text>
  </View>
</View>
`;
